src/main/java/ro/unibuc/triplea/infrastructure/game/repository/GameRepositoryImpl.java

```diff
@@ -0,0 +1,132 @@
+package ro.unibuc.triplea.infrastructure.game.repository;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Primary;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestTemplate;
+
+import com.fasterxml.jackson.databind.JsonNode;
+
+import ro.unibuc.triplea.application.auth.dto.response.GameResponse;
+import ro.unibuc.triplea.domain.auth.model.entity.Game;
+import ro.unibuc.triplea.domain.auth.repository.GameRepository;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+@Component
+@Primary
+public class GameRepositoryImpl implements GameRepository {
+
+    @Value("${STEAM_API_KEY}")
+    private String steamApiKey;
+    
+    private final RestTemplate restTemplate;
+    private final SpringDataGameRepository springDataGameRepository;
+
+    public GameRepositoryImpl(RestTemplate restTemplate, SpringDataGameRepository springDataGameRepository) {
+        this.restTemplate = restTemplate;
+        this.springDataGameRepository = springDataGameRepository;
+    }
+
+    @Override
+    public List<GameResponse> findAll(Optional<Integer> count) {
+        String apiUrl = "http://api.steampowered.com/ISteamApps/GetAppList/v0002/?key=" + steamApiKey + "&format=json";
+        ResponseEntity<JsonNode> responseEntity = restTemplate.getForEntity(apiUrl, JsonNode.class);
+
+        if (responseEntity.getStatusCode().is2xxSuccessful()) {
+
+            JsonNode responseNode = responseEntity.getBody();
+
+            List<GameResponse> games = new ArrayList<>();
+            if (responseNode != null && responseNode.has("applist")) {
+                JsonNode appList = responseNode.get("applist");
+                if (appList.has("apps")) {
+                    JsonNode apps = appList.get("apps");
+
+                    int limit = count.orElse(Integer.MAX_VALUE);
+                    int counter = 0;
+                    for (JsonNode app : apps) {
+                        if (counter >= limit) {
+                            break;
+                        }
+
+                        if (app.get("name").asText() != "") {
```

Comment:
string should not be compared with !=. Use .equals() instead or .equalsIgnoreCase()

Reply:
To avoid NPEs we could also switch the order of the comparison, so that it won't try to find methods of a null object and cause an error:
`if !("".equals(app.get("name").asText()))`

Reply:
fixed